cmake/CMakeLists.txt

```diff
@@ -1365,15 +1365,17 @@ if (onnxruntime_ENABLE_TRAINING)
 
     find_package(MPI)
 
-    if (MPI_CXX_FOUND)
-      message( STATUS "MPI Version: ${MPI_CXX_VERSION}")
-      message( STATUS "MPI (include: ${MPI_CXX_INCLUDE_DIRS}, library: ${MPI_CXX_LIBRARIES})" )
-      mark_as_advanced(MPI_CXX_INCLUDE_DIRS MPI_CXX_LIBRARIES)
-      list(APPEND onnxruntime_EXTERNAL_LIBRARIES ${MPI_CXX_LIBRARIES} ${MPI_CXX_LINK_FLAGS})
-    else ()
-      set(onnxruntime_USE_NCCL OFF)
-      set(onnxruntime_USE_MPI OFF)
-      message( WARNING "MPI is not found. Please define onnxruntime_MPI_HOME to specify the path of MPI. Otherwise, NCCL will be disabled." )
+    if (onnxruntime_USE_MPI OR onnxruntime_USE_NCCL)
+      if (MPI_CXX_FOUND)
+        message( STATUS "MPI Version: ${MPI_CXX_VERSION}")
+        message( STATUS "MPI (include: ${MPI_CXX_INCLUDE_DIRS}, library: ${MPI_CXX_LIBRARIES})" )
+        mark_as_advanced(MPI_CXX_INCLUDE_DIRS MPI_CXX_LIBRARIES)
+        list(APPEND onnxruntime_EXTERNAL_LIBRARIES ${MPI_CXX_LIBRARIES} ${MPI_CXX_LINK_FLAGS})
+      else ()
+        set(onnxruntime_USE_NCCL OFF)
+        set(onnxruntime_USE_MPI OFF)
+        message( WARNING "MPI is not found. Please define onnxruntime_MPI_HOME to specify the path of MPI. Otherwise, NCCL will be disabled." )
```

Comment:
I know this is existing behavior. Just wondering whether we should fail the build if MPI not found. It can be confusing if the USE_MPI is set but we internally set it to false and move on, as the warning can easily go unnoticed.